Sources/core/objects/Human.cpp

```diff
@@ -82,6 +84,274 @@ bool Human::TryMove(Dir direct)
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_)
+        {
+            if (!CanTouch(pulled_object_))
```

Comment:
It is bad idea to not move the pulled object if it is in touch range.
Imagine next thing:
```
wwww
dxow
woow
wwww
```
Room 2x2 (`d` - door, `x` - some non-passable object, `w` - wall, `o` - floor). You can always touch the object in this room -> you cannot move it via pulling -> you cannot access the door.

Diagonal movement is the root of all evil. It is better to just do not care about it. If someone will pull item from the corner he just will move it once and after the pull connection will be broken. So it will be needed to start pull from sides of the object in order to establish reliable pull connection. It allow to make implementation relatively simple - it will be easy to calculate direction for pulled object with help of `VDirToDir` function.